### PR TITLE
[#576] add option to disable server-system-logger container

### LINKED ISSUE #576: Expose configuration that allows for the logging sidecar to be disabled
The issues described in https://github.com/datastax/cass-operator/issues/325 discuss the delays in stopping pods that can be observed, particularly when performing restarts due to backup/restore cycles.

Here we're proposing as a manual means of relief, the ability to disable the sidecar entirely - this might primarily be used in testing scenarios where we don't want to take for long restart cycles over and over again.

### PR DESCRIPTION
**Pull request by jsanda**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #576

**Checklist**
- [x] Changes manually tested
- [x] Chart versions updated (if necessary)
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ Changelog for K8ssandra, new PRs should update the `main / unreleased` section w
 When cutting a new release of the parent `k8ssandra` chart update the `main / unreleased` heading to the tag being generated and date `## vX.Y.Z - YYYY-MM-DD` and create a new placeholder section for  `main / unreleased` entries.
 
 ## main / unreleased
+* [ENHANCEMENT] #576 Add option to disable Cassandra loggin sidecar
```
Comment (JeremiahDJordan) [inline CHANGELOG.md:15]: logging?

Comment (jsanda) [reply CHANGELOG.md]: fixed 😄

**Review by JeremiahDJordan — APPROVED** [review]
looks right to me.  disableSystemLoggerSidecar is in the operator CRD.
https://github.com/datastax/cass-operator/blob/master/charts/cass-operator-chart/templates/customresourcedefinition.yaml#L94-L97

**Review by JeremiahDJordan — APPROVED** [review]

File: charts/k8ssandra/templates/cassandra/cassdc.yaml [hunk 2]
```diff
@@ -116,6 +116,9 @@ spec:
         - "-Dcassandra.system_distributed_replication_dc_names={{ $datacenter.name }}"
         - "-Dcassandra.system_distributed_replication_per_dc={{ min 5 $datacenter.size }}"
 {{- end }}
+{{- end }}
+{{- if not .Values.cassandra.loggingSidecar.enabled }}
+  disableSystemLoggerSidecar: true
```
Comment (jdonenine) [inline charts/k8ssandra/templates/cassandra/cassdc.yaml:121]: Do we need the inverse here or will the full resource get overwritten anytime there's a modification to our values?

**Review by jdonenine — APPROVED** [review]
I didn't test it out, just one quick question I had. Have a good weekend!

**Review by emerkle826 — APPROVED** [review]

**Pull request merged**